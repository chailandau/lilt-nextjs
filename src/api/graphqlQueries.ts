import { gql } from 'graphql-request';

import {
    ACCORDION_FRAGMENT,
    CONVERSION_PANEL_FRAGMENT,
    CTA_FRAGMENT,
    FEATURED_MEDIA_FRAGMENT,
    FEATURE_GRID_FRAGMENT,
    HEADING_FRAGMENT,
    HERO_FRAGMENT,
    ICON_TILE__GRID_FRAGMENT,
    IMAGE_FRAGMENT,
    MENU_ITEMS_FRAGMENT,
    META_FRAGMENT,
    PROCESS_FRAGMENT,
    RICH_TEXT_FRAGMENT,
    SLUG_FRAGMENT,
    SPECIALS_FRAGMENT,
    SWITCHBACK_FRAGMENT,
    TEXT_GRID_FRAGMENT,
    TILE_GRID_FRAGMENT
} from './graphqlFragments';

export const PAGES_QUERY = gql`
query PagesQuery {
    Pages {
        ${SLUG_FRAGMENT}
    }
}
`;

export const LEGAL_PAGES_QUERY = gql`
query PagesQuery {
    LegalPages {
        ${SLUG_FRAGMENT}
    }
}
`;

export const PAGE_CONTENT_QUERY = gql`
query PageContentQuery($slug: String!) {
    Pages(where:{ slug: { equals: $slug }}) {
        docs {
            id
            slug
            title
            ${META_FRAGMENT}
            pageSections {
                enableGrid
                gridColor
                enableShapes
                shapes {
                    id
                    shape
                    location
                }
                blocks {
                    ${HERO_FRAGMENT}
                    ${ACCORDION_FRAGMENT}
                    ${CONVERSION_PANEL_FRAGMENT}
                    ${FEATURE_GRID_FRAGMENT}
                    ${FEATURED_MEDIA_FRAGMENT}
                    ${ICON_TILE__GRID_FRAGMENT}
                    ${PROCESS_FRAGMENT}
                    ${SPECIALS_FRAGMENT}
                    ${SWITCHBACK_FRAGMENT}
                    ${TEXT_GRID_FRAGMENT}
                    ${TILE_GRID_FRAGMENT}
                }
            }
            
        }
    }
}
`;

export const LEGAL_PAGE_CONTENT_QUERY = gql`
query LegalPageContentQuery($slug: String!) {
    LegalPages(where:{ slug: { equals: $slug }}) {
        docs {
            id
            title
            slug
            updatedAt
            legalBlocks {
                ${HEADING_FRAGMENT}
                ${RICH_TEXT_FRAGMENT}
            }
        }
    }
}
`;

export const HEADER_QUERY = gql`
query HeaderQuery  {
    Header {
        menuItems {
            ${MENU_ITEMS_FRAGMENT}
            submenuItems {
                ${MENU_ITEMS_FRAGMENT}
            }
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const FOOTER_QUERY = gql`
query FooterQuery {
    Footer {
        socialLinks {
            icon {
                ${IMAGE_FRAGMENT}
            }
            url
            id
        }
        menuItems {
            ${MENU_ITEMS_FRAGMENT}
        }
        copyrightText
    }
}
`;
