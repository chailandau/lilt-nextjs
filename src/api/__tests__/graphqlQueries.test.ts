import { addMocksToSchema } from '@graphql-tools/mock';
import { makeExecutableSchema } from '@graphql-tools/schema';
import { graphql } from 'graphql';

import { mockDefaults } from '../../__mocks__/graphqlQueries.mocks';
import * as graphqlQueries from '../graphqlQueries';
import * as graphqlSchema from '../graphqlSchema.gql';

describe('graphqlQueries', () => {
    const queries = {
        ...graphqlQueries
    };

    Object.entries(queries).forEach(([name, query]) => {
        it(`mocks ${name} successfully`, async () => {
            const schema = makeExecutableSchema({ typeDefs: graphqlSchema });

            const schemaWithMocks = addMocksToSchema({
                schema,
                mocks: mockDefaults
            });

            const result = await graphql({
                schema: schemaWithMocks,
                source: query
            });
            expect(result).toMatchSnapshot();
        });
    });
});
