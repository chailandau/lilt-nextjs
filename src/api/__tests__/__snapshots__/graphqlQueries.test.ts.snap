// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphqlQueries mocks FOOTER_QUERY successfully 1`] = `
{
  "data": {
    "Footer": {
      "copyrightText": "Hello World",
      "menuItems": [
        {
          "externalLink": "Hello World",
          "id": "Hello World",
          "internalLink": {
            "id": "Hello World",
            "slug": "Hello World",
            "title": "Hello World",
          },
          "label": "Hello World",
          "linkType": "external",
        },
        {
          "externalLink": "Hello World",
          "id": "Hello World",
          "internalLink": {
            "id": "Hello World",
            "slug": "Hello World",
            "title": "Hello World",
          },
          "label": "Hello World",
          "linkType": "external",
        },
      ],
      "socialLinks": [
        {
          "icon": {
            "alt": "Hello World",
            "height": 100,
            "id": "Hello World",
            "url": "Hello World",
            "width": 100,
          },
          "id": "Hello World",
          "url": "Hello World",
        },
      ],
    },
  },
}
`;

exports[`graphqlQueries mocks HEADER_QUERY successfully 1`] = `
{
  "data": {
    "Header": {
      "callToAction": {
        "color": "blue",
        "externalLink": "Hello World",
        "id": "Hello World",
        "internalLink": {
          "id": "Hello World",
          "slug": "Hello World",
          "title": "Hello World",
        },
        "label": "Hello World",
        "linkType": "internal",
      },
      "menuItems": [
        {
          "externalLink": "Hello World",
          "id": "Hello World",
          "internalLink": {
            "id": "Hello World",
            "slug": "Hello World",
            "title": "Hello World",
          },
          "label": "Hello World",
          "linkType": "external",
          "submenuItems": [
            {
              "externalLink": "Hello World",
              "id": "Hello World",
              "internalLink": {
                "id": "Hello World",
                "slug": "Hello World",
                "title": "Hello World",
              },
              "label": "Hello World",
              "linkType": "external",
            },
            {
              "externalLink": "Hello World",
              "id": "Hello World",
              "internalLink": {
                "id": "Hello World",
                "slug": "Hello World",
                "title": "Hello World",
              },
              "label": "Hello World",
              "linkType": "external",
            },
          ],
        },
        {
          "externalLink": "Hello World",
          "id": "Hello World",
          "internalLink": {
            "id": "Hello World",
            "slug": "Hello World",
            "title": "Hello World",
          },
          "label": "Hello World",
          "linkType": "external",
          "submenuItems": [
            {
              "externalLink": "Hello World",
              "id": "Hello World",
              "internalLink": {
                "id": "Hello World",
                "slug": "Hello World",
                "title": "Hello World",
              },
              "label": "Hello World",
              "linkType": "external",
            },
            {
              "externalLink": "Hello World",
              "id": "Hello World",
              "internalLink": {
                "id": "Hello World",
                "slug": "Hello World",
                "title": "Hello World",
              },
              "label": "Hello World",
              "linkType": "external",
            },
          ],
        },
      ],
    },
  },
}
`;

exports[`graphqlQueries mocks PAGE_CONTENT_QUERY successfully 1`] = `
{
  "errors": [
    [GraphQLError: Variable "$slug" of required type "String!" was not provided.],
  ],
}
`;

exports[`graphqlQueries mocks PAGES_QUERY successfully 1`] = `
{
  "data": {
    "Pages": {
      "docs": [
        {
          "id": "Hello World",
          "slug": "Hello World",
          "title": "Hello World",
        },
        {
          "id": "Hello World",
          "slug": "Hello World",
          "title": "Hello World",
        },
      ],
    },
  },
}
`;
